hbase-common/src/main/java/org/apache/hadoop/hbase/nio/RefCnt.java

```diff
@@ -34,14 +33,30 @@
   private Recycler recycler = ByteBuffAllocator.NONE;
 
   /**
-   * Create an {@link RefCnt} with an initial reference count = 1. If the reference count become
-   * zero, the recycler will do nothing. Usually, an Heap {@link ByteBuff} will use this kind of
-   * refCnt to track its life cycle, it help to abstract the code path although it's not really
-   * needed to track on heap ByteBuff.
+   * Usually, an Heap {@link ByteBuff} will use this kind of refCnt to track its life cycle,
+   * it help to abstract the code path although it's not really needed to track on heap ByteBuff.
    */
-  public static RefCnt create() {
-    return new RefCnt(ByteBuffAllocator.NONE);
-  }
+  public static final RefCnt NONE = new RefCnt(ByteBuffAllocator.NONE) {
```

Comment:
Is this right ?  before we will create a RefCnt which will still increase/decrement the its reference count value, but with a NONE de-allocator to deallocate the memory (means donothing).... now we will create a RefCnt with reference count value to be 1, and never change... I'm afraid that if upper layer depends one the ref cnt value, then will be messed up..

Reply:
The RefCnt#NONE is mainly to avoid the creation of temporary objects, but your worries are necessary, will revert it.